scripts/python-skywater-pdk/skywater_pdk/cell_list.py

```diff
@@ -0,0 +1,159 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 SkyWater PDK Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import json
+import os
+import pathlib
+import pprint
+import sys
+
+from typing import Tuple, List, Dict
+
+debug = False
+
+# using a list-table here to allow for easier line breaks in description
+rst_template ="""List of cells
+-------------
+
+.. list-table::
+   :header-rows: 1
+
+   * - Cell name
+     - Description
+     - Type
+     - Verilog name
+{cell_list}
+"""
+
+cell_template = """   * - {cell_name}
+     - {description}
+     - {type}
+     - {verilog_name}
+"""
+
+def collect(library_dir) -> Tuple[str, List[str]]:
+    """Collect the available definitions for cells in a library
+
+    Parameters
+    ----------
+    library_dir: str or pathlib.Path
+        Path to a library.
+
+    Returns
+    -------
+    lib : str
+        Library name
+
+    cells : list of pathlib.Path
+        definition files for cells in the library.
+    """
+
+    if not isinstance(library_dir, pathlib.Path):
+        library_dir = pathlib.Path(library_dir)
+
+    libname = None
+    cells = set()
+
+    for p in library_dir.rglob("definition.json"):
+        if not p.is_file():
+            continue
+        cells.add(p)
+        if libname is None:
+            with open(str(p), "r") as sample_json:
+                sample_def = json.load(sample_json)
+                libname = sample_def['library']
+
+    assert len(libname) > 0
+    cells = list(sorted(cells))
+    return libname, cells
+
+
+def generate_rst(library_dir, library_name, cells):
+    """Generate the RST file containing basic information about cells
+
+    Parameters
+    ----------
+    library_dir: str or pathlib.Path
+        Path to a library.
+
+    library_name: str
+        Name of the library
+
+    cells: list of pathlib.Path
+        List of paths to JSON description files
+
+    """
+
+    if not isinstance(library_dir, pathlib.Path):
+        library_dir = pathlib.Path(library_dir)
+
+    file_name = "cell-list.rst"
+
+    cell_list_file = pathlib.Path(library_dir, file_name)
+
+    cell_list = ""
+    for cell in cells:
+        with open(str(cell), "r") as c:
+            cell_json = json.load(c)
+            cell_list = cell_list + cell_template.format(
+                cell_name = cell_json['name'],
+                description = cell_json['description'].replace("\n", "\n       "),
```

Comment:
Look at the `textwrap` module.

Reply:
not that helpful here as the initial indent is different. It would need to be something like
```
d = textwrap.indent(cell_json['description'], "     - ")
description = d[:3] + '*' + d[4:]
```

Based on `textwrap.indent` I can change it to:
```
def indent(text, initial_indent, subsequent_indent):
    def prefixed_lines():
        s = text.splitlines(True)
        yield initial_indent + s[0]
        for line in s[1:]:
            yield subsequent_indent + line
    
    return ''.join(prefixed_lines())

description = indent(cell_json['description'], '   * - ', '     - ')
```

Reply:
bah, my mistake, it can be just textwrap + lstrip...